### PR TITLE
Add validation for age

### PR DESCRIPTION
**Pull request by janhavisinghh**
https://www.pivotaltracker.com/n/projects/2358313/stories/169420263

### CODE REVIEW

**Comment by sanchita** [comment]
@janhavisinghh Please name the pull request as per the story.
Also add the pivotol link to this story in description.

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 1]
```diff
@@ -317,11 +315,11 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
           showEmptyStateError(false)
         }
 
-        BOTH_DATEOFBIRTH_AND_AGE_ABSENT -> {
+        BOTH_DATEOFBIRTH_AND_AGE_ABSENT, AGE_IS_INVALID -> {
           showAgeEmptyError(false)
         }
 
-        INVALID_DATE_OF_BIRTH, DATE_OF_BIRTH_IN_FUTURE -> {
+        INVALID_DATE_OF_BIRTH, DATE_OF_BIRTH_IN_FUTURE, DATE_OF_BIRTH_IS_INVALID_AGE -> {
```
[the remark this answers is not in the document]
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt]: As discussed, this is not valid anymore.

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 2]
```diff
@@ -317,11 +315,11 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
           showEmptyStateError(false)
         }
 
-        BOTH_DATEOFBIRTH_AND_AGE_ABSENT -> {
+        BOTH_DATEOFBIRTH_AND_AGE_ABSENT, AGE_IS_INVALID -> {
```
[the remark this answers is not in the document]
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt]: As discussed, this is not valid anymore.

File: app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/UserInputAgeValidator.kt [hunk 3]
```diff
@@ -23,4 +37,16 @@ class UserInputAgeValidator {
...
+        return when {
+            parsedDate < nowDate.minusYears(120) -> DateIsInvalid
+            else -> DateIsValid(parsedDate)
+        }
+    }
+
+    @VisibleForTesting
+    fun dateInUserTimeZone(): LocalDate {
+        return LocalDate.now(userClock)
+    }
```
[the remark this answers is not in the document]
Comment (janhavisinghh) [reply app/src/main/java/org/simple/clinic/widgets/ageanddateofbirth/UserInputAgeValidator.kt]: Done.

**Pull request merged**